api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -336,6 +336,12 @@ func (c *imageBuilder) createKanikoJobSpec(project mlp.Project, model *models.Mo
 		Component: models.ComponentImageBuilder,
 		Stream:    project.Stream,
 		Team:      project.Team,
+		Labels:    models.MergeProjectVersionLabels(project.Labels, version.Labels),
+	}
+
+	annotations := make(map[string]string)
+	if !c.config.SafeToEvict {
+		annotations["cluster-autoscaler.kubernetes.io/safe-to-evict"] = fmt.Sprint(c.config.SafeToEvict)
```

Comment:
Super nit: this can just be:

```golang
annotations["cluster-autoscaler.kubernetes.io/safe-to-evict"] = "false"
```

Also, perhaps we can leave an inline comment on why we add this annotation conditionally, linking to the [doc](https://kubernetes.io/docs/reference/labels-annotations-taints/#cluster-autoscaler-kubernetes-io-safe-to-evict)?